contracts/ForwarderV4.sol

```diff
@@ -0,0 +1,353 @@
+   *
+   * @param _operator The address which called `safeTransferFrom` function
+   * @param _from The address of the sender
+   * @param _tokenId The token id of the nft
+   * @param data Additional data with no specified format, sent in call to `_to`
+   */
+  function onERC721Received(
+    address _operator,
+    address _from,
+    uint256 _tokenId,
+    bytes memory data
+  ) external virtual override returns (bytes4) {
+    if (autoFlush721) {
+      IERC721 instance = IERC721(msg.sender);
+      require(
+        instance.supportsInterface(type(IERC721).interfaceId),
+        'The caller does not support the ERC721 interface'
+      );
+      /// this won't work for ERC721 re-entrancy
+      instance.safeTransferFrom(address(this), parentAddress, _tokenId, data);
+    }
+
+    return this.onERC721Received.selector;
+  }
+
+  /**
+   * @notice Method to allow for calls to other contracts. This method can only be called by the allowed address
+   * @param target The target contract address whose method needs to be called
+   * @param value The amount of Ether to be sent
+   * @param data The calldata to be sent
+   */
+  function callFromAllowedAddress(
+    address target,
+    uint256 value,
+    bytes calldata data
+  ) external onlyAllowedAddress returns (bytes memory) {
+    (bool success, bytes memory returnedData) = target.call{ value: value }(
+      data
+    );
+    require(success, 'Allowed address call execution failed');
+
+    return returnedData;
+  }
+
+  /**
+   * @inheritdoc IERC1155Receiver
+   */
+  function onERC1155Received(
+    address _operator,
+    address _from,
+    uint256 id,
+    uint256 value,
+    bytes calldata data
+  ) external virtual override returns (bytes4) {
+    IERC1155 instance = IERC1155(msg.sender);
+    require(
+      instance.supportsInterface(type(IERC1155).interfaceId),
+      'The caller does not support the IERC1155 interface'
+    );
+
+    if (autoFlush1155) {
+      instance.safeTransferFrom(address(this), parentAddress, id, value, data);
+    }
+
+    return this.onERC1155Received.selector;
+  }
+
+  /**
+   * @inheritdoc IERC1155Receiver
+   */
+  function onERC1155BatchReceived(
+    address _operator,
+    address _from,
+    uint256[] calldata ids,
+    uint256[] calldata values,
+    bytes calldata data
+  ) external virtual override returns (bytes4) {
+    IERC1155 instance = IERC1155(msg.sender);
+    require(
+      instance.supportsInterface(type(IERC1155).interfaceId),
+      'The caller does not support the IERC1155 interface'
+    );
+
+    if (autoFlush1155) {
+      instance.safeBatchTransferFrom(
+        address(this),
+        parentAddress,
+        ids,
+        values,
+        data
+      );
+    }
+
+    return this.onERC1155BatchReceived.selector;
+  }
+
+  /**
+   * @inheritdoc IForwarder
+   */
+  function flushTokens(address tokenContractAddress)
+    external
+    virtual
+    override
+    onlyAllowedAddress
```

Comment:
why can't we enable flush by any external address?

Reply:
@evanzbitgo @mullapudipruthvik , My concern is any body can send a token implementing the ERc-20 interface but have some malicious code in the transfer function, Will that cause issues like when invoking flushing, the forwarder will invoke the transfer function of the malicious code which could have some possible security issues (Not sure on it)?

Reply:
We should ask this question to the auditors. If we can enable flush from any external address, then it allows for a lot of flexibility and reduction in smart contract code as well